src/core/regimes.rkt

```diff
@@ -230,13 +230,13 @@
   ;; Vectors used to determine if our current alt is better than out running
   ;; best alt.
   (define best-alt-idx (make-vector number-of-points))
-  (define best-alt-cost (make-vector number-of-points))
+  (define best-alt-cost (make-flvector number-of-points))
 
   (for ([point-idx (in-range 0 number-of-points)])
    ;; Set and fill temporary vectors with starting data
    ;; #f for best index and positive infinite for best cost
    (vector-fill! best-alt-idx #f)
-   (vector-fill! best-alt-cost +inf.f)
+   (set! best-alt-cost (make-flvector number-of-points +inf.0))
```

Comment:
This does allocate. Hopefully, being in the outer loop, it's not too expensive...

Reply:
hmm I will see if I can try something else.